src/com/amazon/ion/benchmark/ParseAndCompareBenchmarkResults.java

```diff
@@ -99,8 +99,44 @@ private static BigDecimal getThresholdScore(IonList rawDataList) {
             IonDecimal score = (IonDecimal) rawDataList.get(i);
             rawData.add(score.bigDecimalValue());
         }
-        BigDecimal threshold = calculateDifference(Collections.max(rawData), Collections.min(rawData));
-        return threshold;
+        Collections.sort(rawData);
+        return calculateThreshold(rawData);
+    }
+
+    /**
+     * Using IQR(interquartile range) method to calculate the threshold of detecting regression.
+     * IQR method is used for labeling outliers. The interquartile range is the range between the first (Q1) and the third quartiles (Q3).
+     * Any data lying outside Q3+1.5*IQR or Q1-1.5*IQR is considered as an outlier (IQR is defined as Q3 - Q1).
+     * Regression detection is equivalent as outlier detection.
+     * The relative difference between the edge of outlier detection range can be defined as the threshold of detecting regression.
+     * @param sortedRawData represents a list of sorted BigDecimal which represents the raw values of benchmark results.
+     * @return the calculated threshold value.
+     */
+    public static BigDecimal calculateThreshold(List<BigDecimal> sortedRawData) {
+        int length = sortedRawData.size();
+        BigDecimal Q1 = median(sortedRawData, 0, length / 2 - 1);
+        BigDecimal Q3 = median(sortedRawData, (length + 1) / 2, length - 1);
+        BigDecimal IQR = Q3.subtract(Q1);
+        BigDecimal range = IQR.multiply(BigDecimal.valueOf(1.5));
+        BigDecimal lowerFence = Q1.subtract(range);
+        BigDecimal upperFence = Q3.add(range);
+        return calculateDifference(upperFence, lowerFence);
```

Comment:
Is this what we want? Or do we want to compare the final result against the upper and lower fences to make sure it's between them? Does this somehow achieve that in a way that I'm not seeing?

Reply:
> Is this what we want? Or do we want to compare the final result against the upper and lower fences to make sure it's between them? Does this somehow achieve that in a way that I'm not seeing?

We are not trying compare the final result to lower and upper fences to see whether it falls into the range. The final result represent the relative difference between the benchmark results from two different commits. However, the lower and upper fences define the acceptable data fluctuate range of benchmark result from one commit.  In this way, lower and upper fences will help us to define the acceptable data difference among the raw data set. If the final results bigger than the relative difference calculated from lower fence and upper fence (which can be calculated by calling method `calculateDifference(upperFence, lowerFence)`), we will flag this comparison as regression detected since the final results already out of the acceptable relative difference.